src/corvus_python/word_document_generator/word_document_generator.py

```diff
@@ -0,0 +1,80 @@
+import base64
+from io import BytesIO
+from os import PathLike
+from typing import IO, Dict
+from docxtpl import DocxTemplate, InlineImage
+from docx.shared import Mm
+
+
+class WordDocumentGenerator:
+    """A class for generating Word documents from templates."""
+
+    def render_doc_as_bytes(self, template: DocxTemplate, context: Dict[str, any]) -> bytes:
+        """Render a Word docx document from the provided template and context,
+        and return the result as a byte array.
+
+        Args:
+            template (DocxTemplate): The template to render. Use `get_template` to get the template object.
+            context (Dict[str, any]): The context to render the template with.
+
+        Returns:
+            bytes: The rendered document as a byte array.
+        """
+
+        template.render(context)
+
+        # save to byte array
+        stream = BytesIO()
+        template.save(stream)
+        result = stream.getvalue()
+
+        # dispose of the stream
+        stream.close()
```

Comment:
I think it's good practice to put this `stream.close()` in a `finally` block to ensure it gets called.